src/App.tsx

```diff
@@ -17,18 +22,34 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                setFilteringMode={setFilteringMode}
+                setSearchQuery={setSearchQuery}
+                searchQuery={searchQuery}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {isLoading && <Loader />}
+              <TodoList
+                setIsLoading={setIsLoading}
+                setInspectedTodo={setInspectedTodo}
+                filteringMode={filteringMode}
+                searchQuery={searchQuery}
+                inspectedTodo={inspectedTodo}
+              />
```

Comment:
You can use ternary operator and display either loader or todo list

Reply:
this breaks the data fetching, as TodoList never gets mounted

I've instead hidden the table header in TodoList when _todos_ is empty, giving a similar visual effect while still mounting the component necessary for fetching data